ResearchKit/Common/ORKPasscodeViewController.h

```diff
@@ -0,0 +1,89 @@
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKStepViewController.h"
+#import "ORKDefines.h"
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ The passcode delegate protocol declares methods which forward the success or failure of passcode modification
+ or authentication.
+ */
+ORK_AVAILABLE_DECL
+@protocol ORKPasscodeDelegate <NSObject>
+
+@required
+/**
+ Notifies the delegate that the user has finished authenticating or editing the passcode.
+ 
+ @param viewController      The `ORKPasscodeStepViewController` object in which the passcode input is entered.
+ */
+- (void)passcodeViewControllerDidFinishWithSuccess:(UIViewController *)viewController;
+
+/**
+ Notifies the delegate that the user failed authentication.
+ 
+ @param viewController      The `ORKPasscodeStepViewController` object in which the passcode input is entered.
+ */
+- (void)passcodeViewControllerDidFailAuthentication:(UIViewController *)viewController;
+
+@optional
+/**
+ Notifies the delegate that the user hit the cancel button item. The cancel button is only visible if this method
+ is implemented.
+ 
+ @param viewController      The `ORKPasscodeStepViewController` object in which the passcode input is entered.
+ */
+- (void)passcodeViewControllerDidCancel:(UIViewController *)viewController;
+
+@end
+
+
+/**
+ An `ORKPasscodeStepViewController` object is the view controller for an `ORKPasscodeStep` object.
+ 
+ A passcode view controller can also be instantiated directly for authentication and editing by using one 
+ of the factory methods below. Each factory method requires a delegate to be implemented.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKPasscodeViewController : UINavigationController
+
++ (id)passcodeAuthenticationViewControllerWithText:(nullable NSString *)text
+                                          delegate:(id<ORKPasscodeDelegate>)delegate;
+
++ (id)passcodeEditingViewControllerWithText:(nullable NSString *)text
+                                   delegate:(id<ORKPasscodeDelegate>)delegate;
```

Comment:
I assume the developer want to specify `with touchID` and `passcode type` when let user do editing (for new passcode) ?